### ISSUE TITLE
[ipgen] ip generation errors

### ISSUE THREAD

**Issue by SahkaRum** [lowRISC/opentitan#11249]
generation of rv_plic IP leads to python script error.

> ./util/ipgen.py generate --template-dir hw/ip_templates/rv_plic/ -o test_ip
> 
> Traceback (most recent call last):
> File "./util/ipgen.py", line 149, in <module>
> sys.exit(main())
> File "./util/ipgen.py", line 135, in main
> args.func(ip_template, args)
> File "./util/ipgen.py", line 32, in action_generate
> config_text = config_fp.read()
> AttributeError: 'NoneType' object has no attribute 'read'

**Comment by rswarbrick** [comment]
Thanks for the report! The problem is that you didn't add a config file and we had some missing error checking in the script. The error check is fixed in #11250 but you should be able to get something sensible by running e.g.
```
util/ipgen.py generate --template-dir hw/ip_templates/rv_plic --config-file hw/top_earlgrey/ip_autogen/rv_plic/data/top_earlgrey_rv_plic.ipconfig.hjson -o test_ip
```
Note that our topgen process normally constructs these configurations on the fly, rather than loading them from a file. But we can cheat and grab one of the ones that they dumped out as an example :-)

**Comment by SahkaRum** [comment]
Thanks for response
